lib/services/access_checker.go

```diff
@@ -187,13 +195,29 @@ type AccessChecker interface {
 	// PrivateKeyPolicy returns the enforced private key policy for this role set,
 	// or the provided defaultPolicy - whichever is stricter.
 	PrivateKeyPolicy(defaultPolicy keys.PrivateKeyPolicy) keys.PrivateKeyPolicy
+
+	// GuessIfAccessIsPossible guesses if access is possible for an entire category
+	// of resources.
+	// It responds the question: "is it possible that there is a resource of this
+	// kind that the current user can access?".
+	// GuessIfAccessIsPossible is used, mainly, for UI decisions ("should the tab
+	// for resource X appear"?). Most callers should use CheckAccessToRule instead.
+	GuessIfAccessIsPossible(ctx RuleContext, namespace string, resource string, verb string, silent bool) error
+
+	// CheckAccessToNode checks login access to a given node.
+	CheckLoginAccessToNode(r types.Server, login string, mfa AccessMFAParams) error
+
+	// CheckSessionJoinAccess checks if the identity has access to join the given session.
+	CheckSessionJoinAccess(session types.SessionTracker, sessionOwner *predicate.User, mode types.SessionParticipantMode) error
 }
 
 // AccessInfo hold information about an identity necessary to check whether that
 // identity has access to cluster resources. This info can come from a user or
 // host SSH certificate, TLS certificate, or user information stored in the
 // backend.
 type AccessInfo struct {
+	// Name is the username of the identity.
+	Name string
```

Comment:
Do a grep for `AccessInfo{`, I think there a few places one is created "manually" where you'll need to fill this in

Reply:
👍